bundles/org.openhab.binding.meross/src/main/java/org/openhab/binding/meross/internal/handler/MerossDeviceHandler.java

```diff
@@ -0,0 +1,277 @@
+            updateStatus(ThingStatus.OFFLINE);
+        } else if (status == MerossEnum.OnlineStatus.ONLINE.value()) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    @Override
+    public void setIpAddress(String ipAddress) {
+        if (ipAddress.equals(this.ipAddress)) {
+            return;
+        }
+        this.ipAddress = ipAddress;
+
+        // Check if a valid address
+        try {
+            InetAddress.getByName(ipAddress);
+        } catch (UnknownHostException e) {
+            logger.debug("Trying to set invalid IP address {}", ipAddress);
+            this.ipAddress = null;
+            return;
+        }
+        // Keep the ip address for the next start of the binding
+        Configuration configuration = editConfiguration();
+        configuration.put(PROPERTY_IP_ADDRESS, this.ipAddress);
+        updateConfiguration(configuration);
+    }
+
+    @Override
+    public @Nullable String getIpAddress() {
+        return ipAddress;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (thing.getStatus() == ThingStatus.OFFLINE) {
+            return;
+        }
+        MerossManager manager = this.manager;
+        if (manager == null) {
+            logger.debug("Handling command, manager not available");
+            return;
+        }
+
+        String channelGroup = channelUID.getId().split("\\d+")[0]; // Trailing digits represent the channel if there are
+                                                                   // multiple
+        Namespace namespace = CHANNEL_NAMESPACE_MAP.get(channelGroup);
+        if (namespace == null) {
+            logger.debug("Unsupported channelUID {}", channelUID);
+            return;
+        }
+        int channel = getChannel(channelUID, channelGroup);
+
+        // Local communication is waiting for http call responses, so can take some time. Therefore use the scheduler to
+        // handle any command.
+        scheduler.submit(() -> {
+            try {
+                if (command instanceof RefreshType) {
+                    if (ipAddress == null) {
+                        logger.debug("Not connected locally, refresh not supported");
+                    } else {
+                        // Refresh for most devices is for all data at once (System.All) and not for the specific
+                        // namespace.
+                        // Therefore don't do a refresh if we just had one triggered (probably from another channel) to
+                        // avoid excessive http calls.
+                        long now = System.currentTimeMillis();
+                        if (now > lastRefresh + REFRESH_INTERVAL_MILLIS) {
+                            manager.refresh(Namespace.SYSTEM_ALL);
+                            lastRefresh = now;
+                        }
+                    }
+                    return;
+                }
```

Comment:
else if (command instanceof OnOffType) {
try {
if (OnOffType.ON.equals(command)) {   might need  to be implemented  in the LightHandler class  too but  maybe I am wrong or missing something.  Please tell me what you think about.